### PR TITLE
fix(Videos): Fix videos embed

### LINKED ISSUE #1408: bug(chat): Embedded videos will return error
**Issue:**
When you send a youtube link in chat and go to play it, the video player will show an error 

**OS:**
All

**Screenshot:**
image

### PR DESCRIPTION
**Pull request by lgmarchi**
### 1. Fix videos embed

image

### Which issue(s) this PR fixes 🔨

- Resolve #1408

### CODE REVIEW

File: kit/src/components/embeds/youtube/mod.rs [hunk 1]
```diff
@@ -7,16 +7,39 @@ pub struct Props {
...
+            Some(end_pos) => &video_id_with_params[..end_pos], // Extract the video ID up to the start of additional parameters
+            None => video_id_with_params, // If there is no '&', the entire string is the video ID
+        };
+
+        // Return the video ID
+        return Some(video_id.to_string());
+    }
+
+    // Return None if the URL doesn't start with the expected base embed URL
+    None
```
Comment (dariusc93) [inline kit/src/components/embeds/youtube/mod.rs]: You can try an `Option` if a function returns an `Option`

```suggestion
fn extract_video_id_from_embed_url(embed_url: &str) -> Option<String> {
    let base_embed_url = "https://www.youtube.com/embed/";

    // Using strip_prefix to remove the base URL if it exists
    let video_id_with_params = embed_url.strip_prefix(base_embed_url)?;

    // Now, extract the video ID before any additional parameters
    let video_id = match video_id_with_params.find('&') {
        Some(end_pos) => &video_id_with_params[..end_pos], // Extract the video ID up to the start of additional parameters
        None => video_id_with_params, // If there is no '&', the entire string is the video ID
    };

    // Return the video ID
    Some(video_id.to_string())
```

Comment (lgmarchi) [reply kit/src/components/embeds/youtube/mod.rs]: Thanks, did this change!

**Comment by phillsatellite** [comment]
During testing with Sara we've noticed that links for Youtube appear great but for Youtube Shorts and other social media platforms like Facebook, Instagram, Tik Tok the previews for videos does not show. This was on Windows and m1/m2 Macbook

Screenshot 2024-02-08 at 3 09 06 PM
Screenshot 2024-02-08 at 3 09 15 PM

**Comment by stavares843** [comment]
btw youtube works but youtube shorts don't display the video 🔨 

as we discussed on discord cc @lgmarchi 😂

**Review by stavares843 — APPROVED** [review]

**Comment by stavares843** [comment]
we can fix the youtube shorts in another PR 🔨

**Review by phillsatellite — APPROVED** [review]

**Review by dariusc93 — APPROVED** [review]

**Pull request merged**